sw/host/rom_ext_image_tools/signer/src/main.rs

```diff
@@ -71,128 +82,111 @@ fn parse_hex_str(hex_str: &str) -> Result<Vec<u8>> {
-        key.public_exponent_be().iter().rev().cloned(),
-    )?;
-    image.set_manifest_field(
-        &manifest::ROM_EXT_SIGNATURE_KEY_MODULUS,
-        key.public_modulus_be().iter().rev().cloned(),
-    )?;
-    image.set_manifest_field(
-        &manifest::ROM_EXT_MANIFEST_IDENTIFIER,
-        parse_hex_str("0x4552544f")?,
-    )?;
-
-    image.set_manifest_field(
-        &manifest::ROM_EXT_IMAGE_VERSION,
-        // TODO: Use into_iter once it's stabilized (for all usages in this file).
-        std::array::IntoIter::new(0_u32.to_le_bytes()),
-    )?;
-
-    // TODO: Do we need these fields?
-    let extensions = [
-        (
-            &manifest::ROM_EXT_EXTENSION0_OFFSET,
-            &manifest::ROM_EXT_EXTENSION0_CHECKSUM,
-        ),
-        (
-            &manifest::ROM_EXT_EXTENSION1_OFFSET,
-            &manifest::ROM_EXT_EXTENSION1_CHECKSUM,
-        ),
-        (
-            &manifest::ROM_EXT_EXTENSION2_OFFSET,
-            &manifest::ROM_EXT_EXTENSION2_CHECKSUM,
-        ),
-        (
-            &manifest::ROM_EXT_EXTENSION3_OFFSET,
-            &manifest::ROM_EXT_EXTENSION3_CHECKSUM,
-        ),
-    ];
-    for fields in &extensions {
-        image.set_manifest_field(
-            fields.0,
-            std::array::IntoIter::new(0_u32.to_le_bytes()),
-        )?;
-        image.set_manifest_field(
-            fields.1,
-            std::array::IntoIter::new(0_u32.to_le_bytes()),
-        )?;
-    }
-    image.set_manifest_field(
-        &manifest::ROM_EXT_USAGE_CONSTRAINTS,
-        std::array::IntoIter::new(
-            [0; manifest::ROM_EXT_USAGE_CONSTRAINTS.size_bytes],
-        ),
-    )?;
-    image.set_manifest_field(
-        &manifest::ROM_EXT_PERIPHERAL_LOCKDOWN_INFO,
-        std::array::IntoIter::new(
-            [0; manifest::ROM_EXT_PERIPHERAL_LOCKDOWN_INFO.size_bytes],
-        ),
-    )?;
-    image.set_manifest_field(
-        &manifest::ROM_EXT_IMAGE_TIMESTAMP,
-        std::array::IntoIter::new(0_u64.to_le_bytes()),
-    )?;
-    image.set_manifest_field(
-        &manifest::ROM_EXT_IMAGE_LENGTH,
-        std::array::IntoIter::new(u32::try_from(image.len())?.to_le_bytes()),
-    )?;
+fn update_image_manifest(image: &mut Image, key: &PrivateKey) -> Result<()> {
+    image.manifest.identifier = 0x4552544f;
+    image.manifest.reserved0 = 0;
+    image.manifest.image_length = u32::try_from(image.bytes().len())?;
+    image.manifest.image_version = 0;
+    image.manifest.image_timestamp = 0;
+    const EXPONENT_SIZE: usize = size_of::<u32>();
+    let exponent_be = key.public_exponent_be();
+    ensure!(
+        exponent_be.len() <= EXPONENT_SIZE,
+        "Public exponent ({} bytes) is longer than {} bytes.",
+        exponent_be.len(),
+        EXPONENT_SIZE,
+    );
+    image.manifest.exponent = u32::from_be_bytes(
+        iter::repeat(0)
+            .take(EXPONENT_SIZE - exponent_be.len())
+            .chain(exponent_be.iter().cloned())
+            .collect::<Vec<_>>()
```

Comment:
I dislike this. I understand it, but I dislke it.

Reply:
I was able to simplify this considerably by initializing this field to zero, then copying over bytes using `Iterator::zip`. Thanks for the suggestion.